lightning/src/offers/offer.rs

```diff
@@ -114,6 +114,23 @@ use std::time::SystemTime;
 
 pub(super) const IV_BYTES: &[u8; IV_LEN] = b"LDK Offer ~~~~~~";
 
+/// An identifier for an [`Offer`] built using [`DerivedMetadata`].
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub struct OfferId(Nonce);
```

Comment:
Hmm... we need to extract this from the `InvoiceRequest` in order to have `Bolt12Invoice` construction be stateless. So it would require adding to the `offer_metadata` or changing it in some way. So my inclination is to not allow a custom value.

Reply:
We wouldn't be able to create a `Bolt12Invoice` for such an `Offer` as we'd fail to verify the `InvoiceRequest`, IIUC.